### PR TITLE
enh(twig): Update keyword list

### PR DESCRIPTION
**Pull request by florianjr**
- Update keyword, filter and tag list
- Update metafiles with my personal information

Resolves: #2422

### CODE REVIEW

**Comment by florianjr** [comment]
@joshgoebel here's is now my PR for the keywords. I'll open a dedicated one for the fixes soon :)

File: src/languages/twig.js [hunk 1]
```diff
@@ -28,17 +29,17 @@ export default function(hljs) {
   var FILTER = {
     begin: /\|[A-Za-z_]+:?/,
     keywords:
-      'abs batch capitalize column convert_encoding date date_modify default ' +
-      'escape filter first format inky_to_html inline_css join json_encode keys last ' +
-      'length lower map markdown merge nl2br number_format raw reduce replace ' +
```
Comment (joshgoebel) [inline src/languages/twig.js:33]: Are `markdown` and `filter` removed/deprecated?  Are they still in use in any supported versions?  We try to be conservative in removing keywords if they could potentially still be in use in code snippets, etc.

File: src/languages/twig.js [hunk 2]
```diff
@@ -28,17 +29,17 @@ export default function(hljs) {
...
-      'reverse round slice sort spaceless split striptags title trim upper url_encode',
+      'abs batch capitalize column convert_encoding country_name currency_name currency_symbol data_uri date date_modify default ' +
+      'escape filter first format format_currency format_date format_datetime format_number format_time html_to_markdown inky_to_html inline_css join json_encode keys language_name last ' +
+      'length locale_name lower map markdown_to_html merge nl2br number_format raw reduce replace ' +
+      'reverse round slice sort spaceless split striptags timezone_name title trim u upper url_encode',
```
Comment (joshgoebel) [inline src/languages/twig.js:35]: Please add `|0` to score the `u` as 0 relevancy.

**Review by joshgoebel — CHANGES_REQUESTED** [review]

**Comment by joshgoebel** [comment]
Looking good.  Almost there!

File: src/languages/twig.js (hunk 1 above)
Comment (florianjr) [reply src/languages/twig.js:33]: Yes, they were removed.

File: src/languages/twig.js (hunk 2 above)
Comment (florianjr) [reply src/languages/twig.js]: I don't know if this is what you mean :)

File: src/languages/twig.js (hunk 1 above)
Comment (joshgoebel) [reply src/languages/twig.js:33]: But people are presumable still using Twig 2.0, yes?  The docs are still available.

https://twig.symfony.com/doc/2.x/filters/index.html

Comment (florianjr) [reply src/languages/twig.js:33]: Then I add it again, I just looked at the 3.0. Thanks :)

File: src/languages/twig.js (hunk 2 above)
Comment (joshgoebel) [reply src/languages/twig.js]: Ype!

File: src/languages/twig.js (hunk 1 above)
Comment (joshgoebel) [reply src/languages/twig.js:33]: Wait, I thought `filter` was a tag?

Comment (florianjr) [reply src/languages/twig.js:33]: It's both :)

Comment (florianjr) [reply src/languages/twig.js:33]: How should we classify it?

Comment (florianjr) [reply src/languages/twig.js:33]: Yes, it is. So it's alright.

File: src/languages/twig.js [hunk 3]
```diff
@@ -28,17 +29,17 @@ export default function(hljs) {
...
+      'reverse round slice sort spaceless split striptags timezone_name title trim u|0 upper url_encode',
     contains: [
       FUNCTIONS
     ]
   };
 
-  var TAGS = 'apply autoescape block deprecated do embed extends filter flush for from ' +
-    'if import include macro sandbox set use verbatim with';
+  var TAGS = 'apply autoescape block deprecated do embed extends flush for from ' +
+    'if import in include is macro sandbox set use verbatim with';
```
Comment (joshgoebel) [inline src/languages/twig.js:42]: Sorry I was just looking at these more closely... aren't these keywords, not tags?  I don't see any `inend` or `isend` in the docs?

Comment (florianjr) [reply src/languages/twig.js:42]: I just thought about this. To be precise they're operators, but I think we could handle them as functions.

**Comment by joshgoebel** [comment]
Sorry I'm doing a terrible job reviewing this... let me try again:

Perhaps we need a new OPERATORS array, stuff them in there (for documentation) then decide which rule they go into AFTER that...

> "and", "or", "is empty", "same as","in" etc. not highlighting.

This was from the original issue which say this resolves... but I don't see mention of `and`, or `or` anywhere... does this truly resolve the original issue?  Are those keywords just dead in 3.0 now?

**Comment by florianjr** [comment]
No sorry, I'm just unexperienced. [Here](https://twig.symfony.com/doc/3.x/), under Operators you can see the current ones listed. We should make a new array for this, that's true.

**Comment by joshgoebel** [comment]
> We should make a new array for this, that's true.

Yes, probably should be it's own array and then join it to the TAGS inside the `template-tag` rule on line 59 or so:

```diff
        className: 'template-tag',
        begin: /\{%/, end: /%}/,
        contains: [
          {
            className: 'name',
            begin: /\w+/,
-            keywords: TAGS,
+            keywords: TAGS + " " + OPERATORS.join(" "),
            starts: {
```

Or something.  Not sure if this applies also to variables or not?

**Comment by joshgoebel** [comment]
Although to handle the grouped keywords ("is empty", "same as") you'll likely need your own mode:

```
GROUPED = {
  begin: (\b(is empty|same as)\b)
}
```

To use in addition to the normal more boring keywords...

**Comment by joshgoebel** [comment]
Ping.

**Comment by joshgoebel** [comment]
Closing for now until you have time to circle back and work on this.
